### DISCUSSION TITLE
Parse PDF according to colouring of the text?

### DISCUSSION THREAD

**Discussion by Rapid1898-code** [jsvine/pdfplumber#823] [Ask for help with specific PDFs]
Hello - i have the attached example.pdf and i would like to know if it is possible to read the text
and also get the information which text is coloured in the background with orange / green / blue / purple / etc.?

Is there any way to do this with pdfplumber?

[exmpl.pdf](https://github.com/jsvine/pdfplumber/files/10812123/exmpl.pdf)

**Comment by jsvine** [comment]
@Rapid1898-code Very interesting example. In this case, the coloring doesn't appear to come from the text itself or core PDF contents, but rather from the annotations. You can access/examine those using the `page.annots` attribute.

**Comment by Rapid1898-code** [comment]
Thanks for the respone - i tried to read the annots for the first page - and the first annot has this data

`{'page_number': 1, 'object_type': 'annot', 'x0': 157.532, 'y0': 743.262, 'x1': 438.499, 'y1': 789.522, 'doctop': 52.36779999999999, 'top': 52.36779999999999, 'bottom': 98.6278000000001, 'width': 280.967, 'height': 46.260000000000105, 'uri': None, 'title': 'philipp', 'contents': None, 'data': {'AP': {'N': <PDFStream(133): raw=27, {'BBox': [0.0, 0.0, 280.967, 46.2595], 'FormType': 1, 'Length': 27, 'Matrix': [1.0, 0.0, 0.0, 1.0, 0.0, 0.0], 'Resources': {'ExtGState': {'R0': {'AIS': False, 'CA': 0.399994, 'Type': /'ExtGState', 'ca': 0.399994}, 'R1': {'AIS': False, 'BM': /'Multiply', 'Type': /'ExtGState'}}, 'ProcSet': [/'PDF'], 'XObject': {'MWFOForm': <PDFObjRef:134>}}, 'Subtype': /'Form', 'Type': /'XObject'}>}, 'C': [1.0, 0.819611, 0.0], 'CA': 0.399994, 'CreationDate': b"D:20230216145420+01'00'", 'F': 4, 'M': b"D:20230216145420+01'00'", 'NM': b'24b0dec2-1b11-514d-9876-4a3c48af76ed', 'P': <Page:1>, 'Popup': {'F': 28, 'Open': False, 'Parent': <PDFObjRef:59>, 'Rect': [595.276, 674.161, 799.276, 788.161], 'Subtype': /'Popup', 'Type': /'Annot'}, 'QuadPoints': [169.155, 788.161, 426.876, 788.161, 169.155, 744.623, 426.876, 744.623], 'Rect': [157.532, 743.262, 438.499, 789.522], 'Subj': b'Hervorheben', 'Subtype': /'Highlight', 'T': b'philipp', 'Type': /'Annot'}}`

and this one

`{'page_number': 1, 'object_type': 'annot', 'x0': 595.276, 'y0': 674.161, 'x1': 799.276, 'y1': 788.161, 'doctop': 53.72880000000009, 'top': 53.72880000000009, 'bottom': 167.7288000000001, 'width': 204.0, 'height': 114.0, 'uri': None, 'title': None, 'contents': None, 'data': {'F': 28, 'Open': False, 'Parent': <PDFObjRef:59>, 'Rect': [595.276, 674.161, 799.276, 788.161], 'Subtype': /'Popup', 'Type': /'Annot'}}`

is it somehow possible to detect which color this annot have?
(eg. the first annot on the first page has yellow annot and the second one is orange)

**Reply by jsvine** [reply]
The first kind — `'Subtype': /'Highlight'` — is the one you want. Looks like the `'C': [1.0, 0.819611, 0.0]` attribute there is indicating an RGB-value color.

**Comment by Rapid1898-code** [comment]
Thanks again - shouldn´t be the RGB color allways digit from 0 to 255?
(and not decimal values like 0.xy?)

**Reply by jsvine** [reply]
Not in annotations. See [p. 607 of the PDF spec](https://ghostscript.com/~robin/pdf_reference17.pdf) for more detail:

<img width="619" alt="Screen Shot" src="https://user-images.githubusercontent.com/534702/221647754-5a592f4e-dbc3-4f9e-978a-e20f75d66f9b.png">
